src/backends/CommerceTools.js

```diff
@@ -0,0 +1,115 @@
+import SdkAuth from '@commercetools/sdk-auth'
+import {ProductSelectorError} from '../ProductSelectorError';
+
+export class CommerceTools {
+  constructor({host, projectKey, clientId, clientSecret, apiUrl, scope = 'view_published_products', locale}) {
+
+    this.sdkAuth = new SdkAuth({
+      host,
+      projectKey,
+      disableRefreshToken: false,
+      credentials: {
+        clientId,
+        clientSecret,
+      },
+      scopes: [scope + ':' + projectKey],
+    });
+
+    this.apiUrl = apiUrl;
+    this.locale = locale;
+  }
+
+  async getHeaders() {
+    this.token = await this.sdkAuth.clientCredentialsFlow();
+    const {
+      token_type, access_token
+    } = this.token;
+
+    return {
+      headers: {
+        'Authorization': `${token_type} ${access_token}`
+      }
+    };
+  }
+
+  async getItems(state, filterIds = []) {
+    const {
+      PAGE_SIZE,
+      params: {
+        projectKey
+      }
+    } = state;
+    try {
+      if (!filterIds.length){
+        return [];
+      }
+      const headers = await this.getHeaders();
+      const idsStrings = '"' + filterIds.join('","') + '"';
+
+      const params = {
+        method: 'GET',
+        ...headers
+      };
+      const response = await fetch(`${this.apiUrl}/${projectKey}/product-projections?staged=false&limit=${PAGE_SIZE}&where=id in (${idsStrings})`, params);
+      const {results} = await response.json();
+
+      return this.parseResults(results);
+
+    } catch (e) {
+      console.error(e);
+      throw new ProductSelectorError('Could not get items', ProductSelectorError.codes.GET_SELECTED_ITEMS);
+    }
+  }
+
+  getImage(masterVariant) {
+    const largeImage = masterVariant.attributes && masterVariant.attributes.find(x => x.name === 'largeImageUrl');
+    return (largeImage && largeImage.value) || (masterVariant.images && masterVariant.images.length ? masterVariant.images[0].url : '')
+  }
+
```

Comment:
I find the above a bit hard to read maybe refactor slightly if that's ok.

```javascript
function  getImage({ attributes = [], images = [] }) {
  const LARGE_IMAGE_KEY = 'largeImageUrl'
  const largeImage = attributes.find(({ name = '' }) => name === LARGE_IMAGE_KEY);
  
  if (largeImage && largeImage.value) {
    return largeImage.value;
  }
 
  return images.length ? images[0].url : '';
}
```

Reply:
Done